src/stake/stake_keys.rs

```diff
@@ -0,0 +1,170 @@
+use super::Stake;
+use crate::{Error, Height};
+
+use fuel_crypto::PublicKey;
+use hashbrown::HashMap;
+
+use alloc::borrow::ToOwned;
+use core::cmp;
+use core::ops::{Bound, Range, RangeBounds};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct StakeKeys {
```

Comment:
[nit] Add some rust docs explaining what StakeKeys does. It's not immediately obvious how this differs from `StakePool` without looking into the code/api a bit.

Reply:
The name was indeed super confusing, even tho the `StakeKeys` is only internal.

Renamed them to `ValidatorsStakes` and `HeightStakes`

https://github.com/FuelLabs/fuel-bft/pull/18/commits/932842df3ec03475b86d37eac0628188f35be494

Reply:
(force-push rebasing)